src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,42 @@
+import { useRef, useState } from "react";
+import ModalBasic from "./ModalBasic/ModalBasic";
+import styles from "./Modal.module.scss";
+import { useOnClickOutside } from "usehooks-ts";
+
+interface Modal {
+  name: string;
+  subtitle?: string;
+  shareButtons?: boolean;
+}
+
+function Modal({ name, ...rest }: Modal) {
```

Comment:
props가 많지 않은 경우에는 이렇게 일일이 적어주는게 컴포넌트를 이해하기 더 좋을 것 같아요~
```suggestion
function Modal({ name, subtitle, shareButtons}: Modal) {
```